core/src/main/scala/caliban/execution/Executor.scala

```diff
@@ -163,7 +191,7 @@ object Executor {
 
         val filteredFields    = mergeFields(currentField, objectName)
         val (deferred, eager) =
-          if (isDeferredEnabled) {
+          if (Feature.isDeferEnabled(flags)) {
```

Comment:
Should we also be checking whether `isStreamEnabled` here?

Reply:
No this is only checking for defer at the fragment level, stream can only apply to list types so no need to check here